cognite/client/data_classes/data_modeling/queries.py

```diff
@@ -0,0 +1,343 @@
+        filter: Filter = None,
+        sort: list[InstanceSort] = None,
+        limit: int = None,
+    ):
+        self.from_ = from_
+        self.through = through
+        self.filter = filter
+        self.sort = sort
+        self.limit = limit
+
+    def dump(self, camel_case: bool = False) -> dict[str, Any]:
+        output: Dict[str, Any] = {"nodes": {}}
+        nodes = output["nodes"]
+        if self.from_:
+            nodes["from"] = self.from_
+        if self.through:
+            nodes["through"] = self.through.dump(camel_case=camel_case)
+        if self.filter:
+            nodes["filter"] = self.filter.dump()
+
+        if self.sort:
+            output["sort"] = [s.dump(camel_case=camel_case) for s in self.sort]
+        if self.limit:
+            output["limit"] = self.limit
+
+        return output
+
+
+class EdgeResultSetExpression(ResultSetExpression):
+    def __init__(
+        self,
+        from_: Optional[str] = None,
+        max_distance: Optional[int] = None,
+        direction: Literal["outwards", "inwards"] = "outwards",
+        filter: Optional[Filter] = None,
+        node_filter: Optional[Filter] = None,
+        termination_filter: Optional[Filter] = None,
+        limit_each: Optional[int] = None,
+        sort: list[InstanceSort] = None,
+        post_sort: list[InstanceSort] = None,
+        limit: int = None,
+    ):
+        self.from_ = from_
+        self.max_distance = max_distance
+        self.direction = direction
+        self.filter = filter
+        self.node_filter = node_filter
+        self.termination_filter = termination_filter
+        self.limit_each = limit_each
+        self.sort = sort
+        self.post_sort = post_sort
+        self.limit = limit
+
+    def dump(self, camel_case: bool = False) -> dict[str, Any]:
+        output: Dict[str, Any] = {"edges": {}}
+        edges = output["edges"]
+        if self.from_:
+            edges["from"] = self.from_
+        if self.max_distance:
+            edges["maxDistance" if camel_case else "max_distance"] = self.max_distance
+        if self.direction:
+            edges["direction"] = self.direction
+        if self.filter:
+            edges["filter"] = self.filter.dump()
+        if self.node_filter:
+            edges["nodeFilter" if camel_case else "node_filter"] = self.node_filter.dump()
+        if self.termination_filter:
+            edges["terminationFilter" if camel_case else "termination_filter"] = self.termination_filter.dump()
+        if self.limit_each:
+            edges["limitEach" if camel_case else "limit_each"] = self.limit_each
+
+        if self.sort:
+            output["sort"] = [s.dump(camel_case=camel_case) for s in self.sort]
+        if self.post_sort:
+            output["postSort" if camel_case else "post_sort"] = [s.dump(camel_case=camel_case) for s in self.post_sort]
+        if self.limit:
+            output["limit"] = self.limit
+
+        return output
+
+
+class QueryResult(UserDict):
```

Comment:
Let's add `def get_nodes(result_expression: str) -> NodeList` and `def get_edges(result_expression: str) -> EdgeList` methods here. They can raise runtime errors if you try to use them on the wrong type of result expression. Then the user can get correct typing.